src/reduced_order/pod_adaptation.cpp

```diff
@@ -0,0 +1,197 @@
+#include "pod_adaptation.h"
+
+namespace PHiLiP {
+namespace ProperOrthogonalDecomposition {
+
+using DealiiVector = dealii::LinearAlgebra::distributed::Vector<double>;
+
+template <int dim, int nstate>
+PODAdaptation<dim, nstate>::PODAdaptation(std::shared_ptr<DGBase<dim,double>> &_dg, Functional<dim,nstate,double> &_functional)
+        : functional(_functional)
+        , dg(_dg)
+        , all_parameters(dg->all_parameters)
+        , coarsePOD(std::make_shared<ProperOrthogonalDecomposition::CoarsePOD<dim>>(dg))
+        , finePOD(std::make_shared<ProperOrthogonalDecomposition::FinePOD<dim>>(dg))
+        , fineNotInCoarsePOD(std::make_shared<ProperOrthogonalDecomposition::FineNotInCoarsePOD<dim>>(dg))
+        , mpi_communicator(MPI_COMM_WORLD)
+        , pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+{
+    dealii::ParameterHandler parameter_handler;
+    Parameters::LinearSolverParam::declare_parameters (parameter_handler);
+    this->linear_solver_param.parse_parameters (parameter_handler);
+}
+
+template <int dim, int nstate>
+void PODAdaptation<dim, nstate>::progressivePODAdaptation()
+{
+    getDualWeightedResidual();
+    while(abs(error) > all_parameters->reduced_order_param.adaptation_tolerance){
```

Comment:
If that guaranteed to terminate?

Reply:
It was not, I have fixed this in the most recent commit.